### PR TITLE
[do-not-review] (unless you should. iykyk) 

### PR DESCRIPTION
**Pull request by wuisawesome**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by cadedaniel — COMMENTED** [review]
LGTM, will wait for chen approval then let's merge

**Review by rickyyx — COMMENTED** [review]
<div><br/>via <a href="https://giphy.com/heyarnold/">Hey Arnold</a> on <a href="https://giphy.com/gifs/heyarnold-nicksplat-xT1R9YYeCajBym9Gfu">GIPHY</a></div>

File: src/ray/protobuf/new_monitor.proto [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    // Feedback about whether a resource request was infeasible.
+    ResourceRequest infeasible_request_data = 2;
+    // The unfinished requests in the autoscaler. Ray will only use this
+    // information for observability.
+    PendingNode pending_node = 3;
+  }
+}
+
+
+// ============= Batched approach =======================
```
Comment (cadedaniel) [inline src/ray/protobuf/new_monitor.proto:74]: nit: this line isn't as long as the other section separators

File: src/ray/protobuf/new_monitor.proto [hunk 2]
```diff
@@ -0,0 +1,201 @@
+// ============= Scheduling Status Primitives ====================
+//
+// Following fields represents the Ray state autoscaler interested
+// in.
+
+// Represents anti affinity constraints for a resource requirement.
```
Comment (jjyao) [inline src/ray/protobuf/new_monitor.proto:6]: How is this represented: (anti-)affinity with nodes who has label k1 with value v1 or v2?

```
# Try to schedule to the node with A100/P100 graphics card. If not, schedule to other nodes.
actor_1 = Actor.options(
        scheduling_strategy=node_affinity(label_in("gpu_type", ["A100", "P100"], is_soft=true))
    ).remote()
```

File: src/ray/protobuf/new_monitor.proto [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+message AntiAffinity {
+  string label = 1;
+}
+
+message ResourceBundle {
+  map<string, double> resources = 1;
+  repeated AntiAffinity anti_affinities = 2;
+  repeated string dynamic_labels = 3;
```
Comment (jjyao) [inline src/ray/protobuf/new_monitor.proto:14]: what's this?

File: src/ray/protobuf/new_monitor.proto [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+// All unbundles in a resource request requires gang
+// allocation semantics: they should be
+// fullfilled all or nothing.
+message ResourceRequest {
+  repeated ResourceBundle bundles = 1;
+}
+
+// Cluster constraint represents minimial cluster size requirement,
+// issued through ray.autoscaler.sdk.
+message ClusterSizeConstraint {
```
Comment (jjyao) [inline src/ray/protobuf/new_monitor.proto:30]: Is this for `sdk.request_resources()`?

File: src/ray/protobuf/new_monitor.proto [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  // if the node has joined the cluster (active) or not (still starting up).
+  ALIVE = 1;
+  // A terminal state, indicating the autoscaler believes this node will never
+  // join the cluster.
+  DEAD = 2;
+  // A hint that this node will be downscaled. Ray doesn't get a say in whether
+  // this node will be downscaled. It only has an opportunity to minimize the
+  // blast radius. Preemption, can happen for multiple reasons such as spot
+  // instance failure.
+  PREMPTED = 3;
```
Comment (jjyao) [inline src/ray/protobuf/new_monitor.proto:68]: TO_BE_PREMPTED?

Do we get a notice from aws before the spot instance stops?

File: src/ray/protobuf/new_monitor.proto [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+message AutoscalerNode {
+  bytes node_id = 1;
+  AutoscalerNodeStates node_state = 2;
+  string description = 4;
+}
+
+message PendingNode {
+  string node_type = 1;
+  uint64 pending_count = 2;
+  uint64 sent_count = 3;
```
Comment (jjyao) [inline src/ray/protobuf/new_monitor.proto:80]: what's this?

File: src/ray/protobuf/new_monitor.proto (hunk 4 above)
Comment (scv119) [reply src/ray/protobuf/new_monitor.proto:30]: yeah that's correct.

**Comment by wuisawesome** [comment]
Closing, unify discussion on #34009

**Pull request closed without merging**